extensions/wikia/FacebookClient/scripts/FacebookClient.XFBML.js

```diff
@@ -2,19 +2,34 @@
 
 /**
  * Render FB XML tags emebedded into wikitext
- * This file is loaded via ResourceLoader only when FB XML tags are present on the page
  */
 $(function () {
 	'use strict';
 
 	// load api on page load and on demand
 	mw.hook('wikipage.content').add(function ($content) {
-		$.loadFacebookAPI()
-			.done(function () {
-				$('.sso-login').removeClass('hidden');
 
-				// scan the new content for any fb tags
-				FB.XFBML.parse($content[0]);
-			});
+		if ((xfbmlTagsOnPage())) {
+			if (facebookSDKNotLoaded()) {
+				// XFBML tags rendered automatically when SDK loads
+				$.loadFacebookSDK();
+			} else {
+				renderXFBMLTags();
+			}
+		}
+
+		function xfbmlTagsOnPage() {
+			var numOfXFBMLTags = $content.find('[data-type="xfbml-tag"]').length;
```

Comment:
Yea, I've seen that `!!` syntax in javascript before and never really liked it. It saves a bit of typing, and if you're well acquainted with the language it makes sense, but I think it makes things slightly less clear for the sake of brevity. Idk, could just be that I haven't written enough js to get that operator drilled into my head. If it works for you, I think I'll stick with current solution. I like that it's pretty unambiguous and straightforward to read.